tests/test_types.py

```diff
@@ -2002,3 +2002,40 @@ def test_dumping_passes_context_to_inner_type_dump(self):
         context = object()
         DumpOnly(inner_type).dump('foo', context)
         assert inner_type.dump_context == context
+
+
+class TestTransform:
+    def test_loading_calls_inner_type_load_with_result_of_pre_load(self):
+        inner_type = SpyType()
+        transform = Transform(inner_type, pre_load=lambda foo: 'bar')
+        transformed = transform.load('foo')
+        assert inner_type.loaded == 'bar'
+
+    def test_post_load_calls_value_from_inner_type_load(self):
+        inner_type = SpyType()
+        transform = Transform(inner_type, post_load=lambda foo: foo + 'bar')
+        transformed = transform.load('foo')
+        assert inner_type.loaded == 'foo'
+        assert transform.load('foo') == 'foobar'
+
+    def test_dump_calls_inner_type_dump_with_result_of_pre_dump(self):
+        inner_type = SpyType()
+        transform = Transform(inner_type, pre_dump=lambda foo: 'bar')
+        transformed = transform.dump('foo')
+        assert inner_type.dumped == 'bar'
+
+    def test_post_dump_calls_inner_type_dump_with_result_of_pre_dump(self):
+        inner_type = SpyType()
+        transform = Transform(inner_type, post_dump=lambda foo: foo + 'bar')
+        transformed = transform.dump('foo')
+        assert transform.dump('foo') == 'foobar'
+
+    def test_transform_passes_context_to_inner_type_dump(self):
+        inner_type = SpyType()
+        context = object()
+        transform = Transform(inner_type,
+                              post_load = lambda foo: 'foo',
```

Comment:
Why do you specify these particular arguments here and left the other ones. Either do not specify any of arguments, or specify all of them (I prefer not specifying any arguments)